src/test/java/org/terasology/genome/GenomeNewTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.genome;
+
+import org.junit.Before;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.google.common.base.Function;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.genome.GenomeDefinition;
+import org.terasology.genome.GenomeRegistry;
+import org.terasology.genome.breed.BreedingAlgorithm;
+import org.terasology.genome.breed.MonoploidBreedingAlgorithm;
+import org.terasology.genome.breed.mutator.GeneMutator;
+import org.terasology.genome.breed.mutator.VocabularyGeneMutator;
+import org.terasology.genome.genomeMap.GeneIndexGenomeMap;
+import org.terasology.genome.genomeMap.SeedBasedGenomeMap;
+import org.terasology.registry.In;
+import org.terasology.world.WorldProvider;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * A test to check working of multiple traits with different breeding algorithms. Addition of multiple breeding
+ * algorithms was based on the structure of these tests.
+ */
+class GenomeNewTest {
+    @In
+    WorldProvider worldProvider;
+
+    @In
+    GenomeRegistry genomeRegistry;
```

Comment:
Same here, I don't think this works (not even when using the MTE). The tests probably pass nevertheless as these fields are never used. I'd be in favor of removing dead code (of preparing for some other changes, move it to another PR).

Reply:
same as above. will remove this